cxx/clean_relation_test.cc

```diff
@@ -52,20 +52,56 @@ BOOST_AUTO_TEST_CASE(test_clean_relation) {
   BOOST_TEST(db->N == 0);
   db->incorporate(false);
   BOOST_TEST(db->N == 1);
+}
+
+BOOST_AUTO_TEST_CASE(test_string_relation) {
+  std::mt19937 prng;
+  Domain D1("D1");
+  Domain D2("D2");
 
   DistributionSpec bigram_spec = DistributionSpec("bigram");
-  CleanRelation<std::string> R2("R1", bigram_spec, {&D2, &D3});
+  CleanRelation<std::string> R2("R2", bigram_spec, {&D1, &D2});
   R2.incorporate(&prng, {1, 3}, "cat");
   R2.incorporate(&prng, {1, 2}, "dog");
   R2.incorporate(&prng, {1, 4}, "catt");
   R2.incorporate(&prng, {2, 6}, "fish");
 
+  double lpg __attribute__((unused));
   lpg = R2.logp_gibbs_approx(D2, 2, 0, &prng);
-  R2.set_cluster_assignment_gibbs(D3, 3, 1, &prng);
-  D1.set_cluster_assignment_gibbs(0, 1);
+  R2.set_cluster_assignment_gibbs(D2, 3, 1, &prng);
+  D1.set_cluster_assignment_gibbs(1, 1);
 
   Distribution<std::string>* db2 = R2.make_new_distribution(&prng);
   BOOST_TEST(db2->N == 0);
   db2->incorporate("hello");
   BOOST_TEST(db2->N == 1);
 }
+
+BOOST_AUTO_TEST_CASE(test_unincorporate) {
```

Comment:
`NoisyRelation::unincorporate` calls `CleanRelation::unincorporate`, could you test NoisyRelation's or disable it? (NoisyRelation manages its own `data` map, so I think `NoisyRelation::unincorporate` needs to remove the item from `data`, which it doesn't currently do).

Reply:
Done (and removed it from it's own data field).